jpyinterpreter/src/main/java/ai/timefold/jpyinterpreter/implementors/StackManipulationImplementor.java

```diff
@@ -296,12 +297,22 @@ public static void restoreExceptionTableStack(FunctionMetadata functionMetadata,
         LocalVariableHelper localVariableHelper = stackMetadata.localVariableHelper;
 
         localVariableHelper.readExceptionTableTargetStack(methodVisitor, exceptionBlock.getTargetInstruction());
+
         for (int i = 0; i < exceptionBlock.getStackDepth(); i++) {
             methodVisitor.visitInsn(Opcodes.DUP);
             methodVisitor.visitLdcInsn(i);
             methodVisitor.visitInsn(Opcodes.AALOAD);
-            methodVisitor.visitTypeInsn(Opcodes.CHECKCAST,
-                    stackMetadata.getTypeAtStackIndex(exceptionBlock.getStackDepth() - i - 1).getJavaTypeInternalName());
+
+            if (functionMetadata.pythonCompiledFunction.pythonVersion.isBefore(PythonVersion.PYTHON_3_11) ||
```

Comment:
Do we want to apply this logic for Python versions >= 3.12?

Reply:
Nope, exclusive for Python 3.11. Joy of the inconsistency of Python bytecode.